index.js

```diff
@@ -35,30 +36,42 @@ app.use(bodyParser.urlencoded({ extended: false }))
 // parse application/json
 app.use(bodyParser.json())
 
-async function check_jenkins() {
-  const jenkinsOpts = {};
-  if (config.JENKINS_AUTH) {
+const healthchecks = {
+  jenkins: async function () {
+    const jenkinsOpts = {};
+    if (!config.JENKINS_AUTH) {
+      return { jenkins: 'no_auth' };
+    }
+
     jenkinsOpts.headers = {'Authorization': 'Basic ' + new Buffer.from(config.JENKINS_AUTH, 'utf8').toString('base64')};
+    const response = await fetch(config.JENKINS_HOST + '/whoAmI/api/json', jenkinsOpts)
```

Comment:
how often is this triggered? and will restarting it help at all? (liveness)

Reply:
I mean its configurable, but ~liveness~readyness is for when k8s makes sure the pod is ready. I figure thats the only time it makes sense to do credentials checking.

I guess i should actually rename them, i was intending this to be the startup check

Reply:
startup check isn't available on our version of k8s, readiness check is still called all the time.

Reply:
Really? That seems dumb. I know there are two probes. I've never tried setting them to different things before

Ite probably a good thing to have both and have Prometheus or something else  check the expensive one

Reply:
startup probes are available from 1.16 which should be okay to upgrade to but might be better to wait for olbak 👅 .

and yeah we've been bitten by assuming readiness was only called on startup.